lib/authHandlers.js

```diff
@@ -149,6 +151,25 @@ var sessions       = require('client-sessions'),
     }
   }
 
+  var handleSSOSetup = function(req, res, next) {
+    console.log("EREZ handleSSOSetup: ", req.query.access_token);
+    secondarySessionTokens[req.query.user_token] = req.query.access_token;
+    next();
+  }
+
+  var handleSSOLogin = function(req, res, next) {
+    console.log("EREZ handleSSOLogin: req.query = ", req.query);
+    if (req.query.user_token) {
+      console.log("EREZ handleSSOLogin: req has user token!!");
+      req.query.access_token = secondarySessionTokens[req.query.user_token];
```

Comment:
I think we can simply remove the token from the cache once the new session takes it and insert it into it's session.